# BLAST, orthologs, and paralogs (Week 2, 20 pts possible)

__Name:__

__Student ID:__

*_Please answer the following questions which are each worth 2 points. Be clear and concise with any written answers. Late assignments will be penalized with a 2pt deduction without prior communication of extenuating circumstances._*

__Q1__.  Paste in code and output for the `for loop`  __Exercise 1__ (pluralizing fruit names)

__Q2__. Create a table that indicates blastp runtime as a function of word size.  Is there a linear relationship?  Why does does changing the word size change the search time?

__Q3__.  How much faster is blastp than water at a world size of 5?

__Q4__.  How long will it take to search two proteomes with word length 5?

__Q5__.  Using E-value to determine the best match, find a worm gene with a single fly ortholog and record their names and the reciprocal E-Values.  Is there more than one fly "hit" for your worm gene?  If so how do you decide if there is a single orthlog?

__Q6__.  What logic would you use to find orthologs automatically using BLAST?

__Q7__.  Discuss what makes orthology difficult to determine by the reciprocal BLAST method.

__Q8__.  How many paralogs did you find for T21B10.2a and for B0213.10?  What criteria did you use?

__Q9__.  Discuss what makes paralogy difficult to determine by BLAST.

__Q10__.  Is alignment score sufficient for determining orthology or paralogy? What other sources of information from the BLAST output might be useful?  What other types of analyses would be helpful?

Add and commit changes for both this document and your lab notebook to the repository.  Push to GitHub and set an issue to indicate that you are ready for this to be graded.