---
#layout: lab
title: Working with Illumina Sequence data
hidden: true    <!-- To prevent post from being displayed as regular blog post -->
tags:
- Brassica
- Illumina
- Linux
---

Today we begin a series of five labs that all work with the same data set.  The data is RNA sequencing data generated from [_Brassica rapa_](http://en.wikipedia.org/wiki/Brassica_rapa), a plant with varieties that can grown for turnips, napa cabbage, or oil seeds (like canola).

In these labs we will explore how gene expression is influenced by plant variety, growing conditions, and organ type.  In this experiment we have:

* Two plant varieties:
    * IMB211: a variety with rapid reproduction that has been bred for laboratory work
    * R500: an oil-seed variety
* Multiple growth conditions
    - growth chamber: simulated sun or shade
    - greenhouse: crowded or uncrowded
    - field: crowded or uncrowded
* Multiple tissue types
    - seedling stem (hypocotyl)
    - leaf blade
    - lead petiole
    - root
    - internode
    - inflorescence (young flower stem)
    - silique (fruit)

We won't work with all of these samples through all of the labs, but the final section on genetic networks will make use of most of them.

Over the next several weeks the goals are to:

1. Learn about Illumina reads, how to map them, and quality control (today)
2. How to view reads in a genome browser and how to find single nucleotide polymorphisms (next week)
3. Find genes that are differentially expressed between genotypes or treatments
4. Ask if differentialy expressed genes have any common functionality (Gene ontologies) or promoter motifs
5. Build a gene regulatory network to determine how genes connect to one another.

## Background for today's lab.

The raw output from an Illumina sequencer is a [fastq file](http://en.wikipedia.org/wiki/FASTQ_format).  Illumina sequencers generate 100 million to 200 million sequences of 50 to 250 bp in length per flow-cell "lane".