---
layout: lab
title: Viewing Illumina reads in IGV; finding SNPs
hidden: true    <!-- To prevent post from being displayed as regular blog post -->
tags:
- Brassica
- Illumina
- Linux
---

# Goals

In the last lab period we learned about Illumina reads and fastq files.  We performed quality trimming and then mapped those reads to the _B. rapa_ reference genome.

Today we will pick up where we left off.  Our goals are to:

1. Learn about sequence alignment/mapping (SAM/BAM) files.
2. Examine mapped reads in [IGV--the integrative genomics viewer](https://www.broadinstitute.org/igv/).  This will allow us to see how reads are placed on the genome during the mapping process. 
3. Find polymorphic positions in our sequencing data.
4. Determine the number of read counts per gene, in preparations for differential expression analysis.

# Preliminaries

# Examine tophat output


