---
#layout: lab
title: Working with Illumina Sequence data
hidden: true    <!-- To prevent post from being displayed as regular blog post -->
tags:
- Brassica
- Illumina
- Linux
---

Today we begin a series of six labs that all work with the same data set.  The data is RNA sequencing data generated from [_Brassica rapa_](http://en.wikipedia.org/wiki/Brassica_rapa), a plant with varieties that can be grown for turnips, napa cabbage, or oil seeds (like canola).

In these labs we will explore how gene expression is influenced by plant variety, growth conditions, and organ type.  In this experiment we have:

* Two plant varieties:
    * IMB211: a variety with rapid reproduction that has been bred for laboratory work
    * R500: an oil-seed variety
* Multiple growth conditions
    - growth chamber: simulated sun or shade
    - greenhouse: crowded or uncrowded
    - field: crowded or uncrowded
* Multiple tissue types
    - seedling stem (hypocotyl)
    - leaf blade
    - lead petiole
    - root
    - internode
    - inflorescence (young flower stem)
    - silique (fruit)

We won't work with all of these samples through all of the labs, but the final section on genetic networks will make use of most of them.

Over the next several weeks the goals are to:

1. Learn about Illumina reads, how to map them, and quality control (today)
2. How to view reads in a genome browser and how to find single nucleotide polymorphisms (next week)
3. Find genes that are differentially expressed between genotypes or treatments
4. Ask if differentialy expressed genes have any common functionality (Gene ontologies) or promoter motifs
5. Build a gene regulatory network to determine how genes connect to one another.

## Background for today's lab.

### Fastq files
The raw output from an Illumina sequencer is a [fastq file](http://en.wikipedia.org/wiki/FASTQ_format).  Illumina sequencers generate 100 million to 200 million sequences of 50 to 250 bp in length per flow-cell "lane".

### Illumina Sequencing
You can read more about the technology at [Illumina's website](http://www.illumina.com/content/dam/illumina-marketing/documents/products/illumina_sequencing_introduction.pdf) and you can watch a short video on Illumina sequencing [here](https://www.youtube.com/embed/HMyCqWhwB8E?iframe&rel=0&autoplay=1)

### Multiplexing
Because each lane on a sequencer generates 150-200M reads we often want to run more than one sample per lane.  This is cause multiplexing.  After sequencing we want to assign each read to the sample that it originaly came from.  How?  When libraries are made we include a short "barcode" or "index" on each adapter that indicates the library from which it came.  We can then read the barcode from each sequence to assign it to a particular sample.  

There are two general procedures.  __In line__ barcodes are placed at the ends of the adapter, adjacent to the DNA of interest.  They are the first bases sequenced and form the beginning of the read.  This the type of barcodes that we have in this data set.  In this case the end user (us) must sort the reads according to the barcodes and trim the barcodes off of the DNA sequence.

__Index__ barcodes are internal in the adapter and are read in a separate, short sequencing reaction.  In this case Illumina software will automatically sort the sequence reads for the end-user before delivery.

### Quality Control
The Illumina sequencer assigns a [phred quality score Q](http://en.wikipedia.org/wiki/Phred_quality_score) to each base of sequence.  The quality ranges from 0(low) to 41 (high) and is defined as Q = -10*log10(P) where P is the probability of an error.  So a phred score of 40 corresponds to an error probablity of 1 in 10,000 ( P = 10^-(Q/10) ) In current Illumina data these are encoded as the ascii characters "!"=0 to "J"=41.  See the [fastq wiki](http://en.wikipedia.org/wiki/FASTQ_format#Encoding) for more information.  

### Mapping
Our overall goals with this sequence data are to 

1. Find single nucleotide polymorphisms (SNPs) between the two genotypes, R500 and IMB211
2. Find genes that are differentially expressed.

In order to do either of these we need to know where in the genome the reads map.

There are a number of [read mapping programs available](http://en.wikipedia.org/wiki/List_of_sequence_alignment_software#Short-Read_Sequence_Alignment).  Two of the more popular ones are [bwa](http://bio-bwa.sourceforge.net/) and [bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml).  If you planning on mapping RNAseq reads to a reference genome then bowtie is preferred because it allows reads to be spliced across introns.  If I am mapping RNAseq to cDNAs or genomic sequence data to a genomic reference I usually use BWA, but both should perform well.

## Outline of work

1. Check FASTQ quality
2. Trim reads to keep high-quality reads
3. Split reads based on barcodes
4. Map to reference genome
5. Examine mapped reads in [IGV--the integrative genomics viewer](https://www.broadinstitute.org/igv/)

## Files

We will work with the following file types today

* .fastq -- file of short read data
* .fa -- fasta files for reference genome
* .sam -- [sequence alignment/map file for mapped reads](http://samtools.github.io/hts-specs/SAMv1.pdf)
* .bam -- the binary version of a sam file
* .bai -- index for bam files

